### PR TITLE
Update Edp.pt.xml

### PR DESCRIPTION
**Pull request by ekaris**

### CODE REVIEW

**Comment by ekaris** [comment]
How do I fix this rule?

**Comment by ekaris** [comment]
Need a little help here doing the rewrite @J0WI

**Comment by J0WI** [comment]
See https://github.com/EFForg/https-everywhere/blob/master/ruleset-testing.md for adding test urls and please also add trailing slashes.

**Comment by ekaris** [comment]
Ok, its good now?

File: src/chrome/content/rules/EDP.pt.xml [hunk 1]
```diff
@@ -1,10 +1,13 @@
...
+<ruleset name="Edp.pt">
+	<target host="edp.pt" />
+	<target host="www.edp.pt" />
+	<target host="www.edpessoa.internet.edp.pt" />
+	<rule from="^http://edp\.pt/"
+	to="https://www.edp.pt/" />
+	<test url="http://www.edp.pt/en/Pages/homepage.aspx" />
+	<test url="http://www.edp.pt/pt/Pages/homepage.aspx" />
+	<rule from="^http://www\.edp\.pt/([\w-]+)/Pages/homepage.aspx"
+	to="https://www.edp.pt/$1/Pages/homepage.aspx" />
```
Comment (J0WI) [inline src/chrome/content/rules/EDP.pt.xml]: This is useless, since you rewrite any host below.

**Comment by J0WI** [comment]
There are some more:
edplivebands.edp.pt
comunidade.edp.pt
energia.edp.pt
edponline.edp.pt
1seculodeenergia.edp.pt
provedordocliente.edp.pt
redy.edp.pt
gestaoconsumos.edp.pt
www.edpessoa.internet.edp.pt
fapa.edp.pt
iam.edp.pt
sinergie.misy.eu
sts.edp.pt
ease.edp.pt

geracaoedp.edp.pt (mismatch, self-signed, expired)

Mismatch:
remarks.edp.pt
webmail.edp.pt
(www.)fablabedp.edp.pt
legacy.edp.pt
quemquerserextraordinario.edp.pt (and expired)
(www.)2050.edp.pt

Timeout:
www.voluntariado.edp.pt
(www.)twist.edp.pt
(www.)edpartners.edp.pt
(www.)edpannualreport.edp.pt

Refused:
gi.edp.pt
gslm.edp.pt
gme.edp.pt
geoact.edp.pt
c2c.edp.pt
www.cdme.edp.pt
(www.)browsedp.edp.pt
(www.)premioedpinovacao.edp.pt

**Comment by ekaris** [comment]
It's done except adding a rule to redirect http://www.edp.pt/ (pt or en) /Pages/homepage.aspx to HTTPS can you help @J0WI?

**Comment by J0WI** [comment]
I think it's fine to have a redirect from ^ to https://www and rewrite www to https directly.
So (www.)edp.pt would redirect to https, then to homepage.aspx and last but not least to homepage.aspx over https.

Please do not forget to document the mismatch on ^.

**Comment by ekaris** [comment]
Good?

File: src/chrome/content/rules/EDP.pt.xml [hunk 2]
```diff
@@ -7,6 +7,7 @@ legacy.edp.pt
 quemquerserextraordinario.edp.pt (and expired)
 (www.)2050.edp.pt
 geracaoedp.edp.pt (mismatch, self-signed, expired)
+www.edp.pt on ^
```
Comment (J0WI) [inline src/chrome/content/rules/EDP.pt.xml]: Replace with ^edp.pt. Please also indent the lines.

File: src/chrome/content/rules/EDP.pt.xml [hunk 3]
```diff
@@ -40,6 +41,8 @@ www.cdme.edp.pt
 	<target host="sinergie.misy.eu" />
 	<target host="sts.edp.pt" />
 	<target host="ease.edp.pt" />
+	<target host="www.edp.pt" />
```
Comment (J0WI) [inline src/chrome/content/rules/EDP.pt.xml]: edp.pt is also needed. You may place them on the top of the list.

File: src/chrome/content/rules/EDP.pt.xml [hunk 4]
```diff
@@ -40,6 +41,8 @@ www.cdme.edp.pt
...
 	
-	<rule from="^http:" to="https:" />
+	<rule from="^" to="https://www" />
+	<rule from="www" to="https:www" />
```
Comment (J0WI) [inline src/chrome/content/rules/EDP.pt.xml]: This won't work.
Use `<rule from="^http://edp\.pt/" to="https://www.edp.pt/" />` and `<rule from="^http:" to="https:" />`.

**Comment by ekaris** [comment]
Error on tests

**Comment by J0WI** [comment]
Yes, our tests are currently broken :(
Rule LGTM so far, but please use tabs to style all the comments and redirect rules.

**Comment by ekaris** [comment]
Yes but it has an error on edp.pt rule too

**Comment by ekaris** [comment]
How many tab clicks should I use for the style?

File: src/chrome/content/rules/EDP.pt.xml [hunk 5]
```diff
@@ -1,10 +1,50 @@
...
+	<target host="gestaoconsumos.edp.pt" />
+	<target host="www.edpessoa.internet.edp.pt" />
+	<target host="fapa.edp.pt" />
+	<target host="iam.edp.pt" />
+	<target host="sinergie.misy.eu" />
+	<target host="sts.edp.pt" />
+	<target host="ease.edp.pt" />
+	
+<rule from="^http:" to="https:" />
+<rule from="^http://edp\.pt/" to="https://www.edp.pt/" />
```
Comment (J0WI) [inline src/chrome/content/rules/EDP.pt.xml]: This needs to be before the generic redirect.

**Comment by J0WI** [comment]
You can copy the code style from other rules (e.g. https://github.com/EFForg/https-everywhere/blob/master/src/chrome/content/rules/ThePirateBay.xml)

**Comment by ekaris** [comment]
Good?

**Comment by J0WI** [comment]
LGTM

**Pull request merged**